faraday/deviceconfiguration.py

```diff
@@ -0,0 +1,440 @@
+#!/usr/bin/env python
+
+import time
+import logging.config
+import os
+import sys
+import json
+import ConfigParser
+import base64
+import argparse
+import shutil
+import requests
+
+from flask import Flask
+from flask import request
+
+from faraday.proxyio import faradaybasicproxyio
+from faraday.proxyio import faradaycommands
+from faraday.proxyio import deviceconfig
+
+# Start logging after importing modules
+relpath1 = os.path.join('etc', 'faraday')
+relpath2 = os.path.join('..', 'etc', 'faraday')
+setuppath = os.path.join(sys.prefix, 'etc', 'faraday')
+userpath = os.path.join(os.path.expanduser('~'), '.faraday')
+path = ''
+
+for location in os.curdir, relpath1, relpath2, setuppath, userpath:
+    try:
+        logging.config.fileConfig(os.path.join(location, "loggingConfig.ini"))
+        path = location
+        break
+    except ConfigParser.NoSectionError:
+        pass
+
+logger = logging.getLogger('DeviceConfiguration')
+
+# Create Device Configuration configuration file path
+deviceConfigurationConfigPath = os.path.join(path, "deviceconfiguration.ini")
+faradayConfigPath = os.path.join(path, "faraday_config.ini")
+logger.debug('deviceconfiguration.ini PATH: ' + deviceConfigurationConfigPath)
+logger.debug('faraday_config.ini PATH: ' + faradayConfigPath)
+
+# Load Device Configuration Configuration from deviceconfiguration.ini file
+deviceConfig = ConfigParser.RawConfigParser()
+
+# Command line input
+parser = argparse.ArgumentParser(description='Device Configuration application provides a Flask server to program Faraday radios via an API')
+parser.add_argument('--init-config', dest='init', action='store_true', help='Initialize Device Configuration configuration file')
+parser.add_argument('--init-faraday-config', dest='initfaraday', action='store_true', help='Initialize Faraday configuration file')
+parser.add_argument('--callsign', help='Set Proxy Faraday callsign to connect to and program')
```

Comment:
Changed in https://github.com/FaradayRF/Faraday-Software/pull/219/commits/1556fe743c340aa67beec7e3e4849fb19021df27